src/code.cloudfoundry.org/lib/common/common.go

```diff
@@ -59,3 +60,40 @@ func isRetryableError(err error) bool {
 
 	return false
 }
+
+func IsIPv6Enabled() bool {
+	testAddress := "[::1]:0"
+
+	addr, err := net.ResolveUDPAddr("udp6", testAddress)
+	if err != nil {
+		return false
+	}
+
+	conn, err := net.ListenUDP("udp6", addr)
+	if err != nil {
+		return false
+	}
+	defer conn.Close()
+
+	return true
+}
+
+func ParseIPConfig(conf []net.IP) (net.IP, net.IP) {
+	var containerIP, containerIPv6 net.IP
+
+	for _, ip := range conf {
+		if containerIP == nil && ip.To4() != nil {
+			containerIP = ip
+		}
+
+		if containerIPv6 == nil && ip.To4() == nil && ip.To16() != nil {
+			containerIPv6 = ip
+		}
+
+		if containerIP != nil && containerIPv6 != nil {
+			break
+		}
+	}
+
+	return containerIP, containerIPv6
```

Comment:
same as other comment, containerIP here will always be v4, right? If, so can you rename to be containerIPv4?